File: app/src/main/java/dev/ragnarok/fenrir/mvp/view/IShortedLinksView.java
package dev.ragnarok.fenrir.mvp.view;

import java.util.List;

import dev.ragnarok.fenrir.model.ShortLink;
import dev.ragnarok.fenrir.mvp.core.IMvpView;
import dev.ragnarok.fenrir.mvp.view.base.IAccountDependencyView;


public interface IShortedLinksView extends IAccountDependencyView, IMvpView, IErrorView {
    void displayData(List<ShortLink> links);

    void notifyDataSetChanged();

    void notifyDataAdded(int position, int count);

    void showRefreshing(boolean refreshing);

    void updateLink(String url);

    void showLinkStatus(String status);
}

File: app/src/main/java/dev/ragnarok/fenrir/mvp/view/IShortedLinksView.kt
package dev.ragnarok.fenrir.mvp.view

import dev.ragnarok.fenrir.model.ShortLink
import dev.ragnarok.fenrir.mvp.core.IMvpView
import dev.ragnarok.fenrir.mvp.view.base.IAccountDependencyView

interface IShortedLinksView : IAccountDependencyView, IMvpView, IErrorView {
    fun displayData(links: List<ShortLink>)
    fun notifyDataSetChanged()
    fun notifyDataAdded(position: Int, count: Int)
    fun showRefreshing(refreshing: Boolean)
    fun updateLink(url: String?)
    fun showLinkStatus(status: String?)
}